src/Hocon.Configuration/Config.cs

```diff
@@ -18,82 +20,63 @@ namespace Hocon
     ///     configuration string.
     /// </summary>
     [Serializable]
-    public class Config : HoconRoot, ISerializable, IEquatable<Config>
+    public class Config : HoconObject, ISerializable, IEquatable<Config>
     {
-        private static readonly HoconValue EmptyValue;
-
-        private const string SerializedPropertyName = "_data";
+        /// <summary>
+        ///     Identical to <see cref="HoconConfigurationFactory.Empty" />.
+        /// </summary>
+        /// <remarks>
+        ///     Added for brevity and API backwards-compatibility with Akka.Hocon.
+        /// </remarks>
+        public new static Config Empty { get; } = new Config();
 
-        static Config()
+        public static Config Create(HoconElement root)
         {
-            EmptyValue = new HoconValue(null);
-            EmptyValue.Add(new HoconObject(EmptyValue));
+            return new Config(root);
         }
 
-        [Obsolete("For json serialization/deserialization only", true)]
-        protected Config()
+        private Config()
         {
+            Root = this;
         }
 
         /// <inheritdoc />
         /// <summary>
         ///     Initializes a new instance of the <see cref="Config" /> class.
         /// </summary>
-        protected Config(HoconValue value)
-        {
-            Value = (HoconValue)value.Clone(null);
-            Root = (HoconValue)value.Clone(null);
-        }
-
-        /// <inheritdoc cref="Config(HoconValue)" />
-        protected Config(HoconValue value, Config fallback) : this(value)
-        {
-            MergeConfig(fallback);
-        }
-
-        /// <inheritdoc cref="Config(HoconValue)" />
-        /// <param name="root">The root node to base this configuration.</param>
-        /// <exception cref="T:System.ArgumentNullException">"The root value cannot be null."</exception>
-        public Config(HoconRoot root)
+        public Config(HoconElement root):base(root)
         {
-            Value = (HoconValue)root.Value.Clone(null);
-            Root = (HoconValue)root.Value.Clone(null);
-
-            if (!(root is Config cfg))
-                return;
-
-            foreach(var value in cfg._fallbacks)
+            if(root is Config cfg)
+            {
+                Root = cfg.Root;
+                _fallbacks = cfg._fallbacks.ToList();
```

Comment:
Why is the `ToList()` call necessary here? Can the `cfg._fallbacks` property be modified?

Reply:
Yes, each `Config` instance need to have their own separate instance of fallbacks, else changing (appending a new fallback value in `WithFallback`) will affect ALL `Config` objects before it.